greenplum-db/gpbackup
# Create release pipeline; templatize `gpbackup` pipeline

We are now using Jinja to generate pipeline yml files, since the
gpbackup pipeline (testing against HEAD) and the gpbackup-release
pipeline(testing tagged releases) are very similar.

We now use tag_filtering to ensure that the `gpbackup` pipeline always tests
HEAD against HEAD and the `gpbackup-release` pipeline tests only tagged
versions (of DPM components *and* GPDB versions).
  * DDBoost is an exception to this pattern, as a tagged version of
    DDBoost is required for compilation. The version being tested
    remains consistent, it is only a one-off for the build_binaries job.

Modified nightly-trigger-off to nightly-trigger.
The default behavior is for nightly-trigger to be enabled by default for prod and disabled for dev pipelines

Other changes:
* Add pass condition for build_dependencies in build_binaries
* Remove scale tests from 'All' group
* Remove nightly-triggers & add cache trigger in release-pipeline
* Trigger tests on new gpdb binaries
* Add unique bucket for final github release candidates

Co-authored-by: Junkeun Yi <juyi@pivotal.io>
Co-authored-by: Trevor Yacovone <tyacovone@pivotal.io>
